build/js/intlTelInput.js

```diff
@@ -358,23 +358,26 @@ https://github.com/Bluefieldscom/intl-tel-input.git
             } else if (!$.fn[pluginName].startedLoadingAutoCountry) {
                 // don't do this twice!
                 $.fn[pluginName].startedLoadingAutoCountry = true;
-                var ipinfoURL = "//ipinfo.io";
-                if (this.options.ipinfoToken) {
-                    ipinfoURL += "?token=" + this.options.ipinfoToken;
+                var self = this;
+                if (typeof this.options.geoipProvider === "function") {
+                    this.options.geoipProvider(function(resp) {
+                        self._autoCountryLoaded(resp);
+                    });
                 }
-                // dont bother with the success function arg - instead use always() as should still set a defaultCountry even if the lookup fails
-                $.get(ipinfoURL, function() {}, "jsonp").always(function(resp) {
-                    $.fn[pluginName].autoCountry = resp && resp.country ? resp.country.toLowerCase() : "";
-                    if ($.cookie) {
-                        $.cookie("itiAutoCountry", $.fn[pluginName].autoCountry, {
-                            path: "/"
-                        });
-                    }
-                    // tell all instances the auto country is ready
-                    // TODO: this should just be the current instances
-                    $(".intl-tel-input input").intlTelInput("autoCountryLoaded");
+            }
+        },
+        _autoCountryLoaded: function(country_code) {
+            $.fn[pluginName].autoCountry = country_code.toLowerCase();
+            if ($.cookie) {
+                $.cookie("itiAutoCountry", $.fn[pluginName].autoCountry, {
+                    path: "/"
                 });
             }
+            // tell all instances the auto country is ready
+            // TODO: this should just be the current instances
+            // $(".intl-tel-input input").intlTelInput("autoCountryLoaded");
+            // Why not just call it directly??
+            this.autoCountryLoaded();
```

Comment:
Yea, that makes sense. I thought there would be some reason you weren't doing this to begin with. I have updated the code to remove that.